src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -0,0 +1,512 @@
+            }
+            else
+            {
+                decodeTraceId = data.trace();
+                final OctetsFW payload = data.payload();
+                final DirectBuffer buffer = payload.buffer();
+
+                int offset = payload.offset();
+                int length = payload.sizeof();
+
+                while (length > 0)
+                {
+                    int consumed = decodeState.decode(buffer, offset, length);
+                    offset += consumed;
+                    length -= consumed;
+                }
+            }
+        }
+
+        private void onEnd(
+            EndFW end)
+        {
+            // TODO
+        }
+
+        private void onAbort(
+            AbortFW abort)
+        {
+            // TODO
+        }
+
+        private void onWindow(
+            WindowFW window)
+        {
+            final int replyCredit = window.credit();
+
+            replyBudget += replyCredit;
+            replyPadding += window.padding();
+
+            final int initialCredit = bufferPool.slotCapacity() - initialBudget;
+            doWindow(supplyTraceId.getAsLong(), initialCredit);
+        }
+
+        private void onReset(
+            ResetFW reset)
+        {
+            // TODO
+        }
+
+        private void onSignal(
+            SignalFW signal)
+        {
+            // TODO
+        }
+
+        private void onAmqpHeader(AmqpHeaderFW header)
+        {
+            doAmqpHeader(header.buffer(), header.offset(), header.limit());
+        }
+
+        private void onAmqpOpen(
+            AmqpFrameFW open)
+        {
+            // TODO
+        }
+
+        private void onAmqpBegin(
+            AmqpFrameFW begin)
+        {
+            // TODO
+            /*
+            * get a session from map with channel
+            * */
+        }
+
+        private void onAmqpAttach(
+            AmqpFrameFW attach)
+        {
+            // TODO
+        }
+
+        private void onAmqpFlow(
+            AmqpFrameFW flow)
+        {
+            // TODO
+        }
+
+        private void onAmqpTransfer(
+            AmqpFrameFW transfer)
+        {
+            // TODO
+        }
+
+        private void onAmqpDisposition(
+            AmqpFrameFW disposition)
+        {
+            // TODO
+        }
+
+        private void onAmqpDetach(
+            AmqpFrameFW detach)
+        {
+            // TODO
+        }
+
+        private void onAmqpEnd(
+            AmqpFrameFW end)
+        {
+            // TODO
+        }
+
+        private void onAmqpClose(
+            AmqpFrameFW close)
+        {
+            // TODO
+        }
+
+        private int decodeHeader(
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            final AmqpHeaderFW amqpHeader = amqpHeaderRO.wrap(buffer, offset, offset + length);
+            onAmqpHeader(amqpHeader);
+            decodeState = this::decodeFrame;
```

Comment:
Change the `decodeState` in `onAmqpHeader`, since the next state may vary depending on validity of the AMQP protocol header bytes offered by the client.